### PR TITLE
Agnostic Tools- LOCA exploration tool complete rewrite and custom functionality added

### PR DESCRIPTION
**Pull request by claalmve**
This PR rewrites the LOCA exploration tool in `agnostic.py` to be much more concise and readable in the codebase while also integrating changes for PGE/SCE support for custom arguments passed in (i.e. years, months, metric).

This can be tested with new `cae-notebooks` PR here: https://github.com/cal-adapt/cae-notebooks/pull/89

### CODE REVIEW

**Comment by claalmve** [comment]
**What's left**
- [ ] Add proper documentation for functions

**Review by vicford — APPROVED** [review]
No issues running on my end from a large perspective

File: climakitae/explore/agnostic.py [hunk 1]
```diff
@@ -268,207 +268,256 @@ def create_conversion_function(lookup_tables):
...
+    and runs the passed in metric on all of this data. The results are then returned in 3 values, the first
+    as a dict of statistic names to xr.DataArray single simulation objects (i.e. median),
+    the second as a dict of statistic names to xr.DataArray objects consisting of multiple simulation objects (i.e. middle 10%),
+    and the last as a xr.DataArray of simulations' aggregated values sorted in ascending order.
 
-    # Returning models from stats and aggregated results
-    return (single_model_stats, multiple_model_stats, sorted_sims)
+    Parameters
+    ----------
+    area_subset: str
```
Comment (claalmve) [inline climakitae/explore/agnostic.py:494]: Very open to thoughts about any components of this docstring, particularly around the descriptions of the `area_subset` and `cached_area`.

**Review by elehmer — COMMENTED** [review]
I thought I submitted my comments days ago...sorry

File: climakitae/explore/agnostic.py [hunk 2]
```diff
@@ -268,207 +268,224 @@ def create_conversion_function(lookup_tables):
...
-            sorted_sims == sorted_sims.quantile(0.5, method="nearest")
-        ].simulation.item(),
-        "q3": sorted_sims.loc[
-            sorted_sims == sorted_sims.quantile(0.75, method="nearest")
-        ].simulation.item(),
-        "max": sorted_sims[-1].simulation.item(),
+    metrics = {
+        "Average Max Air Temperature": {
+            "var": "Maximum air temperature at 2m",
+            "agg": np.mean,
```
[the remark this answers is not in the document]
Comment (claalmve) [reply climakitae/explore/agnostic.py:276]: Definitely can and will be. I think that functionality will take more thought in how to create a more customizable metrics, but for the scope of this PR, I kept them to `np.mean` to showcase the capabilities of the tool.

File: climakitae/explore/agnostic.py (hunk 1 above)
Comment (elehmer) [reply climakitae/explore/agnostic.py:494]: I think you explained it well.

**Review by elehmer — APPROVED** [review]

**Comment by claalmve** [comment]
[LOCA run exploration function](https://trello.com/c/1310b87g/522-loca-run-exploration-function)

**Pull request merged**